funding/manager.go

```diff
@@ -2573,14 +2584,32 @@ func makeFundingScript(channel *channeldb.OpenChannel) ([]byte, error) {
 // confirmation, and then to notify the other systems that must be notified
 // when a channel has become active for lightning transactions.
 // The wait can be canceled by closing the cancelChan. In case of success,
-// a *lnwire.ShortChannelID will be passed to confChan.
+// a *lnwire.ShortChannelID will be passed to confChan. If this function
+// detects that the inputs to the funding transaction have been spent by a
+// different transaction, an error will be sent along the spentChan. The
+// spentChan MUST be buffered.
 //
 // NOTE: This MUST be run as a goroutine.
 func (f *Manager) waitForFundingConfirmation(
 	completeChan *channeldb.OpenChannel, cancelChan <-chan struct{},
-	confChan chan<- *confirmedChannel) {
+	confChan chan<- *confirmedChannel, spentChan chan<- error) {
 
 	defer f.wg.Done()
+
+	// If we are the initiator, we'll know the inputs to the funding
+	// transaction and will check for the inputs being used in a different
+	// transaction. This is done before the defer close(confChan) call so
+	// that the select statement in waitForFundingWithTimeout doesn't
+	// accidentally trigger on the closed channel instead of the error
+	// channel.
```

Comment:
Bit of a layering violation to reference the calling function, I think it's enough to just say we don't want to signal on two channels because it will cause races.